Vagrantfile

```diff
@@ -0,0 +1,100 @@
+  config.vm.synced_folder ".", "#{guest_home_dir}/go/src/github.com/kata-containers/tests", type:"rsync"
+  config.vm.synced_folder "#{ENV['GOPATH']}/src/github.com/kata-containers/kata-containers",
+    "#{guest_home_dir}/go/src/github.com/kata-containers/kata-containers", type:"rsync"
+
+  config.vm.provider "libvirt" do |lv|
+    lv.driver = "kvm"
+    lv.cpus = "4"
+    lv.memory = "8192"
+  end
+
+  # Shared provision script.
+  config.vm.provision "shell", env: {"CI_JOB" => job}, inline: <<-SHELL
+
+    export GOPATH="#{guest_home_dir}/go"
+    kata_tests_repo_dir="${GOPATH}/src/github.com/kata-containers/tests"
+
+    env_file="#{guest_home_dir}/ci_job_env"
+    cat <<-EOF > ${env_file}
+export GOPATH="$GOPATH"
+export PATH="/usr/local/go/bin:\$GOPATH/bin:\$PATH"
+export CI="true"
+export CI_JOB="$CI_JOB"
+pushd $kata_tests_repo_dir
+source .ci/ci_job_flags.sh
+popd
+EOF
+
+    source ${env_file}
+
+    # Customize the .bashrc so that it will have the variables exported
+    # after log-in.
+    echo "cd $kata_tests_repo_dir" >> #{guest_home_dir}/.bashrc
+    echo "source $env_file" >> #{guest_home_dir}/.bashrc
+
+    distro=$(source /etc/os-release; echo $ID)
+    case "$distro" in
+      "fedora")
+        # TODO: The kubernetes tests. Maybe it should be handled on
+        # integration/kubernetes/init.sh
+        sudo systemctl stop firewalld || true
+        sudo swapoff -a || true
+
+        # Fedora >= 32 Kernel comes configured with cgroup v2 by default.
+        # This switches back to cgroup v1. It requires a reboot.
+        sudo dnf install -y grubby
+        sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
+
+        # It tries to build a ubuntu based image by default. When it pulls the ubuntu image
+        # from docker.io it can hit the pulling ratio error. Let's use fedora.
+        export osbuilder_distro="fedora"
+        ;;
+      "ubuntu")
+        # TODO: redis-server package fails to install if ipv6 is disabled. Move this
+        # code to the setup script.
+        if [[ $(sudo sysctl -n net.ipv6.conf.all.disable_ipv6) -eq 1 ]]; then
+          sudo sysctl -w net.ipv6.conf.all.disable_ipv6=0
+          sudo sed -i 's/\(net.ipv6.conf.all.disable_ipv6\).*/\1=0/' /etc/sysctl.conf
+        fi
+        sudo systemctl enable ssh
+        ;;
+    esac
+
+    cd ${kata_tests_repo_dir}
+    .ci/setup.sh
+  SHELL
+
+  config.vm.define "fedora", autostart: false do |fedora|
+    fedora.vm.box = "fedora/32-cloud-base"
+    # Fedora is required to reboot so that the change to cgroups v1
+    # makes effect.
+    fedora.vm.provision "shell", reboot: true, inline: <<-SHELL
+      echo "Need to reboot the VM"
+    SHELL
+  end
+
+  config.vm.define "ubuntu", autostart: false do |ubuntu|
+    # TODO: the ubuntu/bionic64 only supports virtualbox.
+    # Decide if we want to support both libvirt and virtualbox providers,
+    # otherwise stick with generic/ubuntu1804
+    #ubuntu.vm.box = "ubuntu/bionic64"
```

Comment:
Commented out code?

Reply:
Yes, I left it here so to explain why I took the decision to not use `ubuntu/bionic64` cloud image. My intention is to remove this commented code before the merge.